### PR TITLE
Add MFA authncontext response validator

### PR DESCRIPTION
**Pull request by pablothedude**
**Be aware this is build on top of #863 and that PR should therefore be mertged back first!**

The validator should check if the IdP has the for the IdP SP
combination configured authn method in the response to the SP when
configured.

The authn method could be found in the authncontextclassref or as value
in the multivalued `http://schemas.microsoft.com/claims/authnmethodsreferences`
attribute.

When not found but configured this should result in an exception.

https://www.pivotaltracker.com/story/show/172718729

### CODE REVIEW

File: library/EngineBlock/Corto/Filter/Command/ValidateMfaAuthnContextClassRef.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+
+use OpenConext\EngineBlock\Assert\Assertion;
+use Psr\Log\LoggerInterface;
+
+/**
+ * The validator will block any incoming IdP assertion were the configured authncontextclassref for a specific CP/IdP combination
+ * doesn't match and will throw an error. As the configured authncontextclassref MUST be set in the authncontextclassref element
+ * OR MUST be set as value in the http://schemas.microsoft.com/claims/authnmethodsreferences attribute.
```
Comment (MKodde) [inline library/EngineBlock/Corto/Filter/Command/ValidateMfaAuthnContextClassRef.php:25]: 1. :+1: for adding this doc block
2. I see the use of lowercased `authncontextclassref` please fix
3. Typo: ~CP/IdP~ SP/IdP

**Comment by MKodde** [comment]
Are you aware that the PHP 72 tests fail on an acceptance test you added in the previous PR?

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Features/MfaAuthncontextclassref.feature [hunk 2]
```diff
@@ -10,22 +10,50 @@ Feature:
       And an Identity Provider named "SSO-IdP"
       And a Service Provider named "SSO-SP"
 
-    Scenario: The configured authncontextclassref should be set if configured with the IdP configuration mapping
-      Given the IdP "SSO-IdP" is configured for authncontextclassref "http://schemas.microsoft.com/claims/multipleauthn" for SP "SSO-SP"
+    Scenario: The configured authn method should be set as authncontextclassref if configured with the IdP configuration mapping
+      Given the IdP "SSO-IdP" is configured for MFA authn method "http://schemas.microsoft.com/claims/multipleauthn" for SP "SSO-SP"
```
Comment (MKodde) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Features/MfaAuthncontextclassref.feature:14]: As mentioned in the previous PR reviews. I'd realy like to see the inconsistently used `authncontextclassref` to the correct camel cased version. Same goes for the filename.

Comment (pablothedude) [reply src/OpenConext/EngineBlockFunctionalTestingBundle/Features/MfaAuthncontextclassref.feature]: Done

**Comment by pablothedude** [comment]
The test was fixed after rebases the changes from master. All other notes were addressed. I'll merge this PR back because no functional changes were requested and I don't want this PR to be blocking. Feel free to discuss IRL when something still needs to be addressed. Then I'll create a separate PR with a appropriate fix.

**Pull request merged**